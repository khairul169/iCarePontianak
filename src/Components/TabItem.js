import React from "react";
import { Text, StyleSheet, TouchableOpacity } from "react-native";
import PropTypes from "prop-types";
import Icon from "./Icon";

const TabItem = ({ icon, active, title, onPress }) => {
  return (
    <TouchableOpacity style={styles.container} onPress={onPress}>
      {icon && (
        <Icon name={icon} size={20} color={active ? "#383838" : "#ccc"} />
      )}
      <Text style={styles.tabTitle}>{title}</Text>
    </TouchableOpacity>
  );
};

TabItem.propTypes = {
  icon: PropTypes.string.isRequired,
  active: PropTypes.bool,
  title: PropTypes.string,
  onPress: PropTypes.func
};

TabItem.defaultProp = {
  active: false
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    paddingVertical: 6
  },
  tabTitle: {
    color: "#626262",
    fontSize: 11,
    marginTop: 1
  }
});

export default TabItem;
