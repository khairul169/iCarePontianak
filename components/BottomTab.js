import React from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import TabItem from "./TabItem";

const BottomTab = ({ navigation, icons }) => {
  const navigateTo = route => {
    navigation.navigate(route);
  };

  return (
    <View style={styles.container}>
      {navigation.state.routes.map((item, index) => (
        <TabItem
          key={index}
          title={item.routeName}
          active={index === navigation.state.index}
          icon={icons[item.key]}
          onPress={() => navigateTo(item.key)}
        />
      ))}
    </View>
  );
};

BottomTab.propTypes = {
  icons: PropTypes.object
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center"
  }
});

export default BottomTab;
