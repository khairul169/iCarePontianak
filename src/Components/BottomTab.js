import React, {useEffect, useState} from 'react';
import {View, StyleSheet, Keyboard} from 'react-native';
import PropTypes from 'prop-types';
import TabItem from './TabItem';

const BottomTab = ({navigation, icons}) => {
  const [keyboardVisible, setKeyboardVisible] = useState(false);

  useEffect(() => {
    const keyboardDidShowListener = Keyboard.addListener(
      'keyboardDidShow',
      () => setKeyboardVisible(true),
    );
    const keyboardDidHideListener = Keyboard.addListener(
      'keyboardDidHide',
      () => setKeyboardVisible(false),
    );
    return () => {
      keyboardDidShowListener.remove();
      keyboardDidHideListener.remove();
    };
  }, []);

  const navigateTo = route => {
    navigation.navigate(route);
  };

  if (keyboardVisible) {
    return <View />;
  }

  return (
    <View style={styles.container}>
      {navigation.state.routes.map((item, index) => (
        <TabItem
          key={index}
          title={item.routeName}
          active={index === navigation.state.index}
          icon={icons[item.key]}
          onPress={() => navigateTo(item.key)}
        />
      ))}
    </View>
  );
};

BottomTab.propTypes = {
  icons: PropTypes.object,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    //borderTopColor: "#CFD8DC",
    //borderTopWidth: 1,
    elevation: 10,
    zIndex: 10,
    backgroundColor: '#fff',
  },
});

export default BottomTab;
