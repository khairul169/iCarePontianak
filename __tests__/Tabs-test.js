import React from 'react';
import renderer from 'react-test-renderer';

import BottomTab from 'components/BottomTab';
import TabItem from 'components/TabItem';

describe('<BottomTab />', () => {
  const defaultProps = {
    navigation: {
      state: {
        routes: [
          {
            key: 'Home',
            routeName: 'Home',
          },
          {
            key: 'Account',
            routeName: 'Account',
          },
        ],
        index: 0,
      },
    },
    icons: {
      Home: 'home',
      Account: 'account',
    },
  };
  const wrapper = renderer.create(<BottomTab {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<TabItem />', () => {
  const defaultProps = {
    title: 'TabItem',
    icon: 'home',
  };
  const wrapper = renderer.create(<TabItem {...defaultProps} />);

  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
