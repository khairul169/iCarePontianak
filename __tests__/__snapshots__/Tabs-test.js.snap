// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomTab /> render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderTopColor": "#CFD8DC",
      "borderTopWidth": 1,
      "flexDirection": "row",
    }
  }
/>
`;

exports[`<TabItem /> render 1`] = `
<View
  accessible={true}
  clickable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "opacity": 1,
      "paddingVertical": 8,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 20,
        },
        Object {
          "color": "#676767",
        },
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï‹œ
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#333",
          "fontSize": 12,
          "marginTop": 2,
        },
        Object {
          "color": "#676767",
        },
      ]
    }
  >
    TabItem
  </Text>
</View>
`;
